File: java/src/main/java/algorithm/utils/errors/shapes/InvalidTrapezoidError.java
package algorithm.utils.errors.shapes;

import java.util.List;

import algorithm.utils.errors.ShapeError;

/**
 * <h1>InvalidTrapezoidError</h1>
 *
 * Exception raised when the provided base lengths cannot form a valid
 * trapezoid.
 *
 * <h2>Example</h2>
 *
 * <pre>
 * <code>
 * import algorithm.utils.errors.shapes.InvalidTrapezoidError;
 *
 * var baseA = 3.0;
 * var baseB = 4.0;
 * var err = new InvalidTrapezoidError(baseA, baseB);
 *
 * System.out.println(err);  // INVALID TRAPEZOID
 * </code>
 * </pre>
 */
public final class InvalidTrapezoidError extends ShapeError {
    /**
     * Initializes a new instance of <code>InvalidTrapezoidError</code>.
     * <p>
     * This constructor accepts two base lengths as `float` values.
     *
     * @param baseA The length of the first base.
     * @param baseB The length of the second base.
     */
    public InvalidTrapezoidError(double baseA, double baseB) {
        super("Trapezoid", List.of(baseA, baseB), "Bases do not form a trapezoid");
    }
}

File: python/src/algorithm/utils/errors/shapes/invalid_trapezoid_error.py
from typing import final

from .. import ShapeError


@final
class InvalidTrapezoidError(ShapeError):
    """
    `InvalidTrapezoidError`
    ==========================

    Exception raised when the provided bases cannot form a valid trapezoid.

    Parameters:
        base_a (float): Length of base a.
        base_b (float): Length of base b.

    Example:
    >>> from algorithm.utils.errors import InvalidTrapezoidError
    >>> raise InvalidTrapezoidError(2, 2)  # Invalid trapezoid
    """

    def __init__(self, base_a: float, base_b: float):
        super().__init__("Trapezoid", [base_a, base_b], "Bases do not form a trapezoid")
